src/Plugin/Phlint.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace PHPCensor\Plugin;
+
+use PHPCensor;
+use PHPCensor\Builder;
+use PHPCensor\Model\Build;
+use PHPCensor\Model\BuildError;
+use PHPCensor\Plugin;
+
+/**
+ * Phlint is a tool with an aim to help maintain quality of php code by analyzing code and pointing out potential code
+ * issues. It focuses on how the code works rather than how the code looks. Phlint is designed from the start to do
+ * deep semantic analysis rather than doing only shallow or stylistic analysis.
+ * https://gitlab.com/phlint/phlint
+ */
+class Phlint extends Plugin
+{
+    /** @var int */
+    protected $allowedErrors = 0;
+
+    /**
+     * @param Builder $builder
+     * @param Build   $build
+     * @param array   $options
+     *
+     * @throws \Exception
+     */
+    public function __construct(Builder $builder, Build $build, array $options = [])
+    {
+        parent::__construct($builder, $build, $options);
+
+        $this->executable = $this->findBinary('phlint');
+
+        if (isset($options['allowed_errors']) && \is_int($options['allowed_errors'])) {
+            $this->allowedErrors = $options['allowed_errors'];
+        }
+    }
+
+    /**
+     * @return bool
+     */
+    public function execute()
+    {
+        $this->builder->executeCommand(
+            'cd "%s" && %s analyze --no-interaction --no-ansi',
+            $this->builder->buildPath,
+            $this->executable
+        );
+
+        // Define that the plugin succeed
+        $success = true;
+
+        $errors = $this->processReport($this->builder->getLastOutput());
+
+        if (0 < \count($errors)) {
+            if (-1 !== $this->allowedErrors && \count($errors) > $this->allowedErrors) {
+                $success = false;
+            }
+
+            foreach ($errors as $error) {
+                $this->build->reportError(
+                    $this->builder,
+                    self::pluginName(),
+                    $error['message'],
+                    BuildError::SEVERITY_HIGH,
+                    $error['file'],
+                    $error['line_from']
+                );
+            }
+        }
+
+        return $success;
+    }
+
+    /**
+     * @return string
+     */
+    public static function pluginName()
+    {
+        return 'phlint';
+    }
+
+    /**
+     * @param string $output
+     * @return array
+     */
+    protected function processReport($output)
+    {
+        \file_put_contents('/var/repos/output', $output);
```

Comment:
left from debugging?

Reply:
Shit.. Yes mate.. I'm gonna remove it once I get home. Thanks!

Reply:
done